crates/turbopack-dev-server/src/update/stream.rs

```diff
@@ -41,28 +40,26 @@ fn extend_issues(issues: &mut Vec<PlainIssueReadRef>, new_issues: Vec<PlainIssue
 async fn get_content_wrapper(
     resource: Value<ResourceIdentifier>,
     get_content: TransientInstance<GetContentFn>,
-) -> Result<ContentSourceResultVc> {
-    let mut content = get_content();
-    while let ContentSourceContent::NeedData(data) = &*content.await?.content.await? {
-        content = data.source.get(
-            &data.path,
-            Value::new(resource_to_data(resource.clone().into_value(), &data.vary)),
-        );
+) -> Result<ContentSourceContentVc> {
+    let mut content = get_content().await?;
+    while let ContentSourceResult::NeedData(data) = &*content {
+        content = data
+            .source
+            .get(
+                &data.path,
+                Value::new(resource_to_data(resource.clone().into_value(), &data.vary)),
+            )
+            .await?;
     }
-    Ok(content)
-}
-
-async fn resolve_static_content(
-    content_source_result: ContentSourceResultVc,
-) -> Result<Option<VersionedContentVc>> {
-    Ok(match *content_source_result.await?.content.await? {
-        ContentSourceContent::NotFound => None,
-        ContentSourceContent::HttpProxy(_) => {
-            panic!("HTTP proxying is not supported in UpdateStream")
-        }
-        ContentSourceContent::Static(content) => Some(content),
-        ContentSourceContent::NeedData(_) => {
-            bail!("this might only happen temporary as get_content_wrapper resolves the data")
+    Ok(match &*content {
+        ContentSourceResult::NeedData(..) => unreachable!(),
```

Comment:
The `while` above should be a `match` within a `loop` to avoid this `unreachable`:

```rust
loop {
    match &*content {
        ContentSourceResult::NeedData(data) => {
          content = ...
        }
        ContentSourceResult::NotFound => return ...,
        ContentSourceResult::Result { .. } => return ...,
    }
}
```